nucypher/cli/actions/configure.py

```diff
@@ -110,3 +118,64 @@ def handle_invalid_configuration_file(emitter: StdoutEmitter,
         emitter.message(INVALID_JSON_IN_CONFIGURATION_WARNING.format(filepath=filepath))
         # ... sorry.. we tried as hard as we could
         raise  # crash :-(
+
+
+def collect_worker_ip_address(emitter: StdoutEmitter, network: str, force: bool = False) -> str:
+
+    # From environment variable  # TODO: remove this environment variable?
+    ip = os.environ.get(NUCYPHER_ENVVAR_WORKER_IP_ADDRESS)
+    if ip:
+        message = f'Using IP address ({ip}) from {NUCYPHER_ENVVAR_WORKER_IP_ADDRESS} environment variable'
+        emitter.message(message, verbosity=2)
+        return ip
+
+    # From node swarm
+    try:
+        message = f'Detecting external IP address automatically'
+        emitter.message(message, verbosity=2)
+        ip = determine_external_ip_address(network=network)
+    except UnknownIPAddress:
+        if force:
+            raise
+        emitter.message('Cannot automatically determine external IP address - input required')
+
+    # Confirmation
+    if not force:
+        if not click.confirm(CONFIRM_URSULA_IPV4_ADDRESS.format(rest_host=ip)):
+            ip = click.prompt(COLLECT_URSULA_IPV4_ADDRESS, type=WORKER_IP)
+
+    validate_worker_ip(worker_ip=ip)
+    return ip
+
+
+def perform_startup_ip_check(emitter: StdoutEmitter, ursula: Ursula, force: bool = False) -> None:
+    """
+    Used on ursula startup to determine if the external
+    IP address is consistent with the configuration's values.
+    """
+    try:
+        external_ip = determine_external_ip_address(network=ursula.domain, known_nodes=ursula.known_nodes)
+    except UnknownIPAddress:
+        message = 'Cannot automatically determine external IP address'
+        emitter.message(message)
+        return  # TODO: crash, or not to crash... that is the question
```

Comment:
What's such a node good for? Wouldn't it have SSL issues for every REST interaction?

Reply:
Perhaps not. This condition does not indicate an invalid IP or mismatch - it just means our efforts to automatically determine it have failed.